editor/src/components/canvas/remix/utopia-remix-root-component.tsx

```diff
@@ -81,17 +100,66 @@ export const UtopiaRemixRootComponent = React.memo((props: UtopiaRemixRootCompon
 
   const routeModules = useGetRouteModules(basePath)
 
+  const [navigationData, setNavigationData] = useAtom(RemixNavigationAtom)
+  const setActiveRemixScene = useSetAtom(ActiveRemixSceneAtom)
+
+  const currentEntries = navigationData[EP.toString(basePath)]?.entries
+  const currentEntriesRef = React.useRef(currentEntries)
+  currentEntriesRef.current = currentEntries
+
   // The router always needs to be updated otherwise new routes won't work without a refresh
   // We need to create the new router with the current location in the initial entries to
   // prevent it thinking that it is rendering '/'
   const router = React.useMemo(() => {
     if (routes == null || routes.length === 0) {
       return null
     }
-
-    return createMemoryRouter(routes)
+    const initialEntries = currentEntriesRef.current == null ? undefined : currentEntriesRef.current
+    return createMemoryRouter(routes, { initialEntries: initialEntries })
   }, [routes])
 
+  const updateNavigationData = React.useCallback(
+    (routerr: RouterType, location: Location) => {
```

Comment:
Can we rename `routerr` to something more sensible please (even `innerRouter` would do)